Source/AwsCommonRuntimeKit/crt/Checksums.swift

```diff
@@ -0,0 +1,31 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import struct Foundation.Data
+import AwsCChecksums
+
+extension Data {
+    
+    /// Computes the CRC32 over data.
+    /// - Parameter previousCrc32:  Pass 0 in the previousCrc32 parameter as an initial value unless continuing to update a running crc in a subsequent call.
+    public func crc32(previousCrc32: UInt32 = 0) -> UInt32 {
+        self.withUnsafeBytes { bufferPointer in
+            guard let ptr = bufferPointer.baseAddress?.assumingMemoryBound(to: UInt8.self) else {
+                return 0
```

Comment:
debatable: I'd prefer we pass `ptr=NULL, len=0` down to the C function. Just use the one canonical implementation, rather than let the language binding handle 1 specific case.